app/src/main/kotlin/model/abstractGraph/Graph.kt

```diff
@@ -80,14 +80,15 @@ abstract class Graph<D> {
 
     fun getVertices() = vertices.toList()
 
+    // This and next two methods are used to localize exceptions
     protected fun getEdge(vertex1: Vertex<D>, vertex2: Vertex<D>): Edge<D> {
         val edge = outgoingEdgesMap[vertex1]?.find { it.isIncident(vertex2) }
-            ?: throw NoSuchElementException("Can't find edge between vertices with ids ${vertex1.id} and ${vertex2.id}")
+            ?: outgoingEdgesMap[vertex2]?.find { it.isIncident(vertex1) }
+            ?: throw NoSuchElementException("No edge between vertices with ids ${vertex1.id} and ${vertex2.id}")
```

Comment:
In a directed graph, if we want to get an edge between (v1) and (v2), we may get an edge between (v2) and (v1), but it is not correct.

I think we need to make this method abstract and implement independently in directed and undirected graphs.

Reply:
Nice remark! Will fix it right now.